src/drake/core.clj

```diff
@@ -176,6 +181,17 @@
       :vars vars
       :opts (if-not method opts (merge (:opts method) opts)))))
 
+(defn- existing-and-empty-inputs
```

Comment:
I think this should return a map like `{:existing (...) :empty (...)}`, so that it will be clearer at the usage site what's going on.